### PR TITLE
docs: add RFC template

### PR DESCRIPTION
**Pull request by waynexia**
Signed-off-by: Ruihang Xia <waynestxia@gmail.com>

# Which issue does this PR close?

Closes #

# Rationale for this change
 
Template for our RFC!

# What changes are included in this PR?

Add `docs/rfcs/`, it contains `text/` for rfc markdown and `resource/` for images.

# Are there any user-facing changes?
no

# How does this change test
no need

### CODE REVIEW

**Comment by ShiKaiWi** [comment]
It seems hard to look up the docs if the file is named after date. Should we name these files according to its content?

File: docs/rfcs/text/YYYYmmDD-template.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
+- Feature Name: (fill me in with a unique ident, my_awesome_feature)
+- Start Date: (fill me in with today's date, YYYY-MM-DD)
```
Comment (jiacai2050) [inline docs/rfcs/text/YYYYmmDD-template.md:2]: Duplicate with title

**Comment by ShiKaiWi** [comment]
What does the .gitkeep do?

File: docs/rfcs/text/YYYYmmDD-template.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
+- Feature Name: (fill me in with a unique ident, my_awesome_feature)
```
Comment (jiacai2050) [inline docs/rfcs/text/YYYYmmDD-template.md:1]: Maybe we can organize rfcs directory like this
```
$ tree
.
├── 20220601-awesome-feature.md
└── resources/
```
It's more easy to refer images in `resources` directory.

Comment (waynexia) [reply docs/rfcs/text/YYYYmmDD-template.md:1]: > It seems hard to look up the docs if the file is named after date. Should we name these files according to its content?

The date is one part of file name. It should follow by feature name, like `20220616-prom_remote_storage.md`

> What does the .gitkeep do?

Empty dir won't be tracked by git. That is a placeholder.

**Comment by waynexia** [comment]
> It seems hard to look up the docs if the file is named after date. Should we name these files according to its content?

The date is one part of file name. It should follow by feature name, like `20220616-prom_remote_storage.md`

> What does the .gitkeep do?

Empty dir won't be tracked by git. That is a placeholder.

File: docs/rfcs/text/YYYYmmDD-template.md (hunk 1 above)
Comment (waynexia) [reply docs/rfcs/text/YYYYmmDD-template.md:2]: I removed "Start Date". "Feature Name" is also included in file name but I think we should keep it?

File: docs/rfcs/text/YYYYmmDD-template.md (hunk 2 above)
Comment (waynexia) [reply docs/rfcs/text/YYYYmmDD-template.md]: 👍

File: docs/rfcs/text/YYYYmmDD-template.md (hunk 1 above)
Comment (jiacai2050) [reply docs/rfcs/text/YYYYmmDD-template.md]: Yeah, title in body is more accurate, filename is short and concise.

**Comment by ShiKaiWi** [comment]
LGTM

**Review by jiacai2050 — APPROVED** [review]
LGTM

**Pull request merged**